src/main/java/decodex/commands/ListCommand.java

```diff
@@ -0,0 +1,41 @@
+package decodex.commands;
+
+import decodex.data.DataManager;
+import decodex.modules.BaseModule;
+import decodex.modules.ModuleManager;
+import decodex.ui.Ui;
+
+public class ListCommand extends Command {
+
+    public static final String COMMAND_WORD = "list";
+
+    public ListCommand() {
+        super();
+    }
+
+    @Override
+    public void run(DataManager dataManager, ModuleManager moduleManager, Ui ui) {
+        BaseModule[] modules = moduleManager.getModules();
+
+        StringBuilder moduleListString = new StringBuilder();
+        int maxNameWidth = 0;
+
+        // Find number of characters of the longest module name
+        for (BaseModule module : modules) {
+            String moduleName = module.getName();
+            if (moduleName.length() > maxNameWidth) {

```

Comment:
Perhaps use `!moduleName.isEmpty()` instead of checking the length is more than 0?

Reply:
I see, then LGTM